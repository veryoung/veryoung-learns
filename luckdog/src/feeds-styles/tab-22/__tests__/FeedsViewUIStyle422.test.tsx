import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle422 from '../FeedsViewUIStyle422';
import { commonProps } from '@test/mocks/props';
import { itemBean } from './mocks/FeedsViewUIStyle422';
import { shallow } from 'enzyme';

describe('<FeedsViewUIStyle422/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        index: 0,
        isEdit: true,
        itemBean: itemBean as any,
        parent: {
          state: {
            isEdit: false,
          },
          mViewModel: {
            deleteExtParam: jest.fn(),
            setExtParams: jest.fn(),
          },
          handleEdit: jest.fn(),
        },
      };
      const tree = renderer.create(<FeedsViewUIStyle422 {...props} />);
      expect(tree).toMatchSnapshot();
    });

    it('click edit', () => {
      const props = {
        ...commonProps,
        index: 0,
        isEdit: true,
        itemBean: itemBean as any,
        parent: {
          state: {
            isEdit: false,
          },
          mViewModel: {
            deleteExtParam: jest.fn(),
            setExtParams: jest.fn(),
          },
          handleEdit: jest.fn(),
        },
      };
      const dom = shallow(<FeedsViewUIStyle422 {...props} />);
      dom.find('.edit').simulate('click');
    });
  });
});
