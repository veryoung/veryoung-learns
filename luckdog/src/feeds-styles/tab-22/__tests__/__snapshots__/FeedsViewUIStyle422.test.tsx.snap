// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle422/> render testing render normal 1`] = `
<div
  nativeName="View"
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }
>
  <div
    nativeName="View"
    onClick={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
      }
    }
  >
    <div
      nativeName="View"
      style={Object {}}
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "rgba(0, 0, 0, 0.5)",
              "rgba(104, 109, 116, 0.5)",
              "rgba(0, 0, 0, 0.5)",
              "rgba(104, 109, 116, 0.5)",
            ],
            "fontSize": 12,
            "lineHeight": 16,
            "paddingBottom": 18,
            "paddingTop": 20,
          }
        }
        text="已加入0本书"
      >
        已加入0本书
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <p
        nativeName="Text"
        style={
          Array [
            Object {
              "colors": Array [
                "rgba(0, 0, 0, 0.5)",
                "rgba(104, 109, 116, 0.5)",
                "rgba(0, 0, 0, 0.5)",
                "rgba(104, 109, 116, 0.5)",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "paddingBottom": 18,
              "paddingTop": 20,
            },
            Object {
              "paddingRight": 12,
            },
          ]
        }
        text="阅读记录"
      >
        阅读记录
      </p>
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "rgba(0, 0, 0, 0.3)",
              "rgba(104, 109, 116, 0.3)",
              "rgba(0, 0, 0, 0.3)",
              "#ffffff",
            ],
            "fontSize": 12,
            "paddingBottom": 19,
            "paddingTop": 19,
          }
        }
        text="|"
      >
        |
      </p>
      <p
        className="edit"
        nativeName="Text"
        onClick={[Function]}
        style={
          Array [
            Object {
              "colors": Array [
                "rgba(0, 0, 0, 0.5)",
                "rgba(104, 109, 116, 0.5)",
                "rgba(0, 0, 0, 0.5)",
                "rgba(104, 109, 116, 0.5)",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "paddingBottom": 18,
              "paddingTop": 20,
            },
            Object {
              "opacity": 1,
              "paddingLeft": 12,
            },
          ]
        }
        text="编辑"
      >
        编辑
      </p>
    </div>
  </div>
  <div
    nativeName="View"
    style={
      Array [
        Object {
          "width": 328,
        },
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        },
      ]
    }
  />
</div>
`;
